router/auth.go

```diff
@@ -136,6 +137,15 @@ func authInjectYellowfin(log *log.Logger, w http.ResponseWriter, req *http.Reque
 		return false
 	}
 
+	// Used to configure the module filter used in YF logins
+	if req.URL.Path == "/yellowfin/setModuleFilter" {
+		var filterValue []byte
```

Comment:
This is the thing I spoke about this morning - ie that filterMap is shared state. There needs to be a lock (mutex) guarding all access to filterMap. As it stands right now, multiple threads will clobber filterMap.

Same as imqsauth, you must treat errors with respect. In this case, the iotil.ReadAll error is being discarded. This is not what we want. Instead, we need to take appropriate action if there is an error there, such as returning a 400 or something. The 400 probably won't reach the client, but it's the right thing to do. Regardless of the response, we don't want to continue processing the request if ioutil.ReadAll returned with an error.